docs/notebooks/trust_region.pct.py

```diff
@@ -0,0 +1,281 @@
+            plt.close(fig)
+
+    gif_file = convert_frames_to_gif(frames)
+    gif = IPython.display.HTML(
+        '<img src="data:image/gif;base64,{0}"/>'.format(
+            base64.b64encode(gif_file.getvalue()).decode()
+        )
+    )
+    IPython.display.display(gif)
+
+
+plot_history(result)
+
+# %% [markdown]
+# ## Batch trust region rule
+#
+# Next we demonstrate how to run Bayesian optimization with the batch trust region rule.
+#
+# ### Create the batch trust region acquisition rule
+#
+# We achieve Bayesian optimization with trust region by specifying `BatchTrustRegionBox` as the
+# acquisition rule.
+#
+# This rule needs an initial number `num_query_points` of sub-spaces (or trust regions) to be
+# provided and performs optimization in parallel across all these sub-spaces. Each region
+# contributes one query point, resulting in each acquisition step collecting `num_query_points`
+# points overall. As the optimization process continues, the bounds of these sub-spaces are
+# dynamically updated.
+#
+# In addition, this is a "meta" rule that requires the specification of a batch aquisition
+# base-rule for performing optimization; for our example we use `EfficientGlobalOptimization`
+# coupled with the `ParallelContinuousThompsonSampling` acquisition function.
+#
+# Note: the number of sub-spaces/regions must match the number of batch query points.
+
+# %%
+num_query_points = 5
+
+init_subspaces = [
+    trieste.acquisition.rule.SingleObjectiveTrustRegionBox(search_space)
+    for _ in range(num_query_points)
+]
+base_rule = trieste.acquisition.rule.EfficientGlobalOptimization(  # type: ignore[var-annotated]
+    builder=trieste.acquisition.ParallelContinuousThompsonSampling(),
+    num_query_points=num_query_points,
+)
+batch_acq_rule = trieste.acquisition.rule.BatchTrustRegionBox(
+    init_subspaces, base_rule
+)
+
+# %% [markdown]
+# ### Run the optimization loop
+#
+# We run the Bayesian optimization loop as before by defining a `BayesianOptimizer` and calling its
+# `optimize` method with the trust region rule. Once the optimization loop is complete, the
+# optimizer will return `num_query_points` new query points for every step in the loop. With
+# 5 steps, that's 25 points in total.
+
+# %%
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+
+num_steps = 5
+result = bo.optimize(
+    num_steps, initial_data, build_model(), batch_acq_rule, track_state=True
+)
+dataset = result.try_get_final_dataset()
+
+# %% [markdown]
+# ### Visualizing batch trust region results
+#
+# We visualize the results as before.
+
+# %%
+plot_final_result(dataset)
+
+# %%
+plot_history(result)
+
+# %% [markdown]
+# ## Trust region `TurBO` acquisition rule
+#
+# Finally, we show how to run Bayesian optimization with the `TurBO` algorithm. This is a
+# trust region algorithm that uses local models and datasets to approximate the objective function
+# within one trust region.
```

Comment:
not sure what the right wording is, but perhaps "within their respective trust regions"? otherwise it could sound like a single TR algorithm? ignore if you think this is sufficiently clear as is

Reply:
I did have slightly different wording before that implied multiple regions, but explicitly changed it to one region. The current TurBO implementation only supports one region. When that is replaced with new classes in a future PR, I can update the wording.